### ISSUE TITLE
request, reduce (dispatcher?) output when VIP interface is not available

### ISSUE THREAD

**Issue by hydrosine** [OpenSIPS/opensips#973]
As mentioned in #972 we use corosync and pacemaker to have opensips highly available.

When the VIP is not on the machine and opensips is running it generates a lot of output from the dispatcher pinging, we would like to be able to reduce this output from opensips.

First the output. It starts 11 seconds after opensips is started.

```
15:09:04 process exited with 0 (opensips started, VIP not there.)
15:09:15 opensips[24055]: ERROR:core:proto_udp_send: sendto(sock,0x7f98cf6adac0,366,0,0x7f98cf6ac268,16): Invalid argument(22)
opensips[24055]: CRITICAL:core:proto_udp_send: invalid sendtoparameters#012one possible reason is the server is bound to localhost and#012attempts to send to the net
opensips[24055]: ERROR:tm:msg_send: send() for proto 1 failed
opensips[24055]: ERROR:tm:t_uac: attempt to send to 'sip:ip.is.he.re:5060' failed
```

this output repeats for every dispatch destination which is to be pinged. (as we have a big list this is a lot of output ;) )

within the same second another PID comes with these messages without an IP, I'm not quite sure what these are within the pinging process.

```
15:09:15 opensips[24061]: ERROR:core:proto_udp_send: sendto(sock,0x7f98cf6adac0,366,0,0x7f98cf6ac268,16): Invalid argument(22)
opensips[24061]: CRITICAL:core:proto_udp_send: invalid sendtoparameters#012one possible reason is the server is bound to localhost and#012attempts to send to the net
opensips[24061]: ERROR:tm:msg_send: send() for proto 1 failed
...
15:09:18 ^^ these messages stop now for a bit.
```

This message also repeats for the amount of dispatcher locations.
Then 15:09:24 the next cycle starts, I think this is the next cycle of pings as our ping_interval is on 10.

To reduce on log clutter I would like to request the first or both of the following solutions:

Have a fifo dispatcher toggle for pinging, so we can disable pinging when the VIP is not active. In our case we could have a script attached to the failover which disables and enables the pinging.

Have a way to tell opensips a listen interface is highly available ie, `listen = udp:ip.is.he.re vip`.
so opensips is aware that the interface is not always there for sending out. and reduce output when it is in this state. This could also come in handy in some other modules which send out messages on their own.

I understand the second solution is something that would be hard to implement and possible has lots of implications ;)

What do you think?

**Comment by bogdan-iancu** [comment]
@hydrosine if you know the server is in "backup" mode, why don't you disable via MI all the destination, so you will prevent OpenSIPS from pinging them. Or at least configure it to so pinging only when is in probing mode (and the switch to probing can be done based on handled calls).

Indeed, having  a way to toggle the pinging may be something interesting.

**Comment by hydrosine** [comment]
Indeed some possible work arounds which I will definitely look into.
Unfortunately disabling all destinations is not really feasible as my set of destinations changes from time to time. Or I should query this from the database which output can then be used to disable them all.

And we want the pinging so we can also do early detection of failure during the calm night hours.

based on your feedback I would like to request the following:
- Wildcard destination activation/deactivation. ```opensipsctl fifo ds_set_state i 0 *``` maybe also  ```opensipsctl fifo ds_set_state i * *.:5060```
    - Before (older versions) we changed the state via database and then reloaded the dispatcher. but this does not work any more for current versions. (DB says active, after ds reload stays Probing.)
    - we already have persistent_state 0

- have an MI command to disable/enable pinging of active gateways.
    ```opensipsctl fifo ds_probing_mode 0```

**Comment by bogdan-iancu** [comment]
This issue is solved starting with 3.0 by using clusterer with sharing tags - see https://opensips.org/html/docs/modules/3.0.x/dispatcher.html#param_cluster_sharing_tag - only the `active` node (according to the tag) will do the pinging.
